src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/ReceivedZipFileConverter.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.services.reports;
+
+import io.vavr.Tuple2;
+import uk.gov.hmcts.reform.bulkscanprocessor.entity.reports.ReceivedZipFile;
+import uk.gov.hmcts.reform.bulkscanprocessor.services.reports.models.ReceivedZipFileData;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toList;
+
+public class ReceivedZipFileConverter {
+
+    public List<ReceivedZipFileData> convertReceivedZipFiles(List<ReceivedZipFile> receivedZipFiles) {
+        return receivedZipFiles
+            .stream()
+            .collect(groupingBy(file ->
+                                    new Tuple2<>(
+                                        file.getZipFileName(),
+                                        file.getContainer()
+                                    )
+            ))
```

Comment:
can we inline somehow? it's a simple operation and entire stack would be so much clearer:

```
stream
collect
values
stream
map
collect
happy days
```